knowledge_repo/app/deploy/gunicorn.py

```diff
@@ -21,12 +21,21 @@ def __init__(self, *args, **kwargs):
         BaseApplication.__init__(self)
 
     def load_config(self):
-        options = {
+        args = self.cfg.parser().parse_args(self.cfg.get_cmd_args_from_env())
+
+        # Load up environment configuration.
+        for key, value in vars(args).items():
+            if key != 'args' and value is not None:
+                self.cfg.set(key, value)
+
+        # Lastly, update the configuration with any command line settings.
+        args = {
             'bind': u'{}:{}'.format(self.host, self.port),
             'workers': self.workers,
-            'timeout': self.timeout
+            'timeout': self.timeout,
         }
-        for key, value in options.items():
+
+        for key, value in args.items():
```

Comment:
do you still need this iteration over `args.items()`, when you're doing the same thing above on line 26?

Reply:
The args here are different. These are the args from the `GUNICORN_CMD_ARGS ` environment variable.